pkg/etcd/etcd.go

```diff
@@ -1067,28 +1077,52 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 			logrus.Warnf("Failed to list nodes with etcd role: %v", err)
 		}
 
+		// a map to track if a node is a member of the etcd cluster or not
+		nodeIsMember := make(map[string]bool)
+		nodesMap := make(map[string]*v1.Node)
+		for _, node := range nodes {
+			nodeIsMember[node.Name] = false
+			nodesMap[node.Name] = node
+		}
+
 		for _, member := range members.Members {
+			var node *v1.Node
+			for _, node = range nodes {
+				if strings.HasPrefix(member.Name, node.Name+"-") {
+					nodeIsMember[node.Name] = true
+				}
+			}
+
 			if member.IsLearner {
 				if err := e.trackLearnerProgress(ctx, progress, member); err != nil {
 					logrus.Errorf("Failed to track learner progress towards promotion: %v", err)
 				}
-				for _, node := range nodes {
-					if strings.HasPrefix(member.Name, node.Name+"-") {
-						_, _, err := e.setEtcdStatusCondition(node, client.CoreV1(), member.Name, false)
-						if err != nil {
-							logrus.Errorf("Unable to set etcd status condition %s: %v", member.Name, err)
-						}
-					}
+
+				if err := e.setEtcdStatusCondition(node, client, member.Name, StatusLearner); err != nil {
+					logrus.Errorf("Unable to set etcd status condition %s: %v", member.Name, err)
 				}
 				break
 			}
 
-			for _, node := range nodes {
-				if strings.HasPrefix(member.Name, node.Name+"-") {
-					_, _, err := e.setEtcdStatusCondition(node, client.CoreV1(), member.Name, true)
-					if err != nil {
-						logrus.Errorf("Unable to set etcd status condition %s: %v", member.Name, err)
-					}
+			// verify if the member is healthy and set the etcd status condition
+			if _, err := e.getETCDStatus(ctx, member.ClientURLs[0]); err != nil {
+				logrus.Errorf("Could not get status for member %s: %v", member.Name, err)
```

Comment:
I think you can drop this log message, since the errors here may include errors extracted from the successful etcd status check.

Could we perhaps include the errors returned by the status check in the status message?

Reply:
Something like this?

```go
func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client kubernetes.Interface, memberName string, memberStatus MemberStatus, errMessage string) error {
	var newCondition v1.NodeCondition
	switch memberStatus {
	case StatusUnhealthy:
		newCondition = v1.NodeCondition{
			Type:    etcdStatusType,
			Status:  "False",
			Reason:  "Unhealthy",
			Message: fmt.Sprintf("Node is unhealthy: %s", errMessage),
		}
```

Reply:
Maybe don't make it specific to errors and the unhealthy status? If it's empty use the default for the condition, if it is non-empty then use what is provided.